File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque
input = sys.stdin.readline

n, m, c = map(int, input().split())

# 인전리스트의 형태로 간선이 연결하는 두 노드 표현
adj = [[] for _ in range(n+1)]
for _ in range(m): # 간선의 개수만큼 반복
    a,b = map(int, input().split())
    adj[a].append(b)
    adj[b].append(a)

for i in range(n+1):
    adj[i].sort()
    
visited = [False] * (n+1)
def dfs(start):
    visited[start] =True
    print(start, end=" ")
    for i in adj[start]:
        if not visited[i]:
            dfs(i)

def bfs(start):
    visited = [False] * (n+1)
    que = deque([start])
    visited[start] = True
    while que:
        v = que.popleft()
        print(v, end=" ")
        for i in adj[v]:
            if not visited[i]:
                que.append(i)
                visited[i] = True

dfs(c)        
print()
bfs(c)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
// let input = require('fs').readFileSync('1260.txt').toString().split('\n');
let input = require('fs').readFileSync('dev/stdin').toString().split('\n');

// 노드의 수, 간선의 수, 시작지점
const [node, edge, start] = input[0].split(' ').map((v) => Number(v));

// 노드의 연결정보를 저장할 인접리스트, node+1개의 빈 배열로 초기화
let graph = [];
for (let i = 0; i <= node; i++) {
  graph.push([]);
}

// 각 노드 별 상호 연결되어 있으므로 아래와 같이 연결해줌
for (let i = 0; i < edge; i++) {
  const [a, b] = input[i + 1].split(' ').map((v) => Number(v));
  graph[a].push(b);
  graph[b].push(a);
}
graph.forEach((e) => {
  e.sort((a, b) => a - b);
});
// console.log(graph);

let visited_dfs = new Array(node + 1).fill(false);
let answer_dfs = [];
function dfs(graph, start) {
  visited_dfs[start] = true;
  answer_dfs.push(start);
  for (let i = 0; i < graph[start].length; i++) {
    if (visited_dfs[graph[start][i]] === false) {
      dfs(graph, graph[start][i]);
    }
  }
}

let visited_bfs = new Array(node + 1).fill(false);
let answer_bfs = [];
function bfs(graph, start) {
  let queue = [start];
  visited_bfs[start] = true;
  answer_bfs.push(start);
  while (queue.length) {
    let v = queue.shift();
    for (let i = 0; i < graph[v].length; i++) {
      if (visited_bfs[graph[v][i]] === false) {
        queue.push(graph[v][i]);
        visited_bfs[graph[v][i]] = true;
        answer_bfs.push(graph[v][i]);
      }
    }
  }
}
dfs(graph, start);
bfs(graph, start);
console.log(answer_dfs.join(' '));
console.log(answer_bfs.join(' '));
